qa/rolling-upgrade/src/test/java/org/elasticsearch/upgrades/TimeSeriesIndexIT.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0 and the Server Side Public License, v 1; you may not use this file except
+ * in compliance with, at your election, the Elastic License 2.0 or the Server
+ * Side Public License, v 1.
+ */
+
+package org.elasticsearch.upgrades;
+
+import org.elasticsearch.client.Request;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.xcontent.XContentBuilder;
+import org.elasticsearch.xcontent.XContentType;
+
+import java.io.IOException;
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.List;
+
+public class TimeSeriesIndexIT extends AbstractRollingTestCase {
+
+    private record DocumentHolder(String timestamp, String host, double gauge, long counter) {}
+
+    private static final String INDEX_NAME = "test";
+    private static final Instant NOW = Instant.ofEpochMilli(System.currentTimeMillis());
+    private static final List<DocumentHolder> DOCUMENTS = List.of(
+        new DocumentHolder(NOW.plus(1, ChronoUnit.MINUTES).toString(), "host-a", 12.0, 3),
+        new DocumentHolder(NOW.plus(2, ChronoUnit.MINUTES).toString(), "host-a", 13.0, 4),
+        new DocumentHolder(NOW.plus(3, ChronoUnit.MINUTES).toString(), "host-a", 15.0, 5),
+        new DocumentHolder(NOW.plus(4, ChronoUnit.MINUTES).toString(), "host-a", 18.0, 10),
+        new DocumentHolder(NOW.plus(5, ChronoUnit.MINUTES).toString(), "host-a", 19.0, 12),
+        new DocumentHolder(NOW.plus(6, ChronoUnit.MINUTES).toString(), "host-a", 22.0, 20),
+        new DocumentHolder(NOW.plus(7, ChronoUnit.MINUTES).toString(), "host-a", 25.0, 30)
+    );
+
+    public void testTimeSeriesIndexSettings() throws IOException {
+        if (CLUSTER_TYPE == ClusterType.OLD) {
+            createTimeSeriesIndex(INDEX_NAME);
+            assertIndexExists(INDEX_NAME);
+            indexDocuments(INDEX_NAME, 0, 5);
+            assertDocCount(client(), INDEX_NAME, 5);
+            assertSearch(INDEX_NAME);
+        } else if (CLUSTER_TYPE == ClusterType.MIXED) {
+            assertIndexExists(INDEX_NAME);
```

Comment:
This test is twice invoked in mixed cluster mode.
I think we should do the indexing once, otherwise when the cluster 2/3 upgraded assertions will fail.
Let's do the indexing here inside an if statement like this: `if (FIRST_MIXED_ROUND) {`?